lib/_io.scm

```diff
@@ -8344,6 +8344,143 @@
 
 ;;;----------------------------------------------------------------------------
 
+(define-prim (##make-raw-device-port raw-device device uname direction)
+  (let ((mutex
+         (macro-make-port-mutex))
+        (rkind
+         (if (##eq? direction (macro-direction-out))
+           (macro-none-kind)
+           (macro-raw-device-kind)))
+        (wkind
+         (if (##eq? direction (macro-direction-in))
+           (macro-none-kind)
+           (macro-raw-device-kind)))
+        (roptions
+         0)
+        (rtimeout
+         #t)
+        (rtimeout-thunk
+         #f)
+        (woptions
+         0)
+        (wtimeout
+         #t)
+        (wtimeout-thunk
+         #f)
+        (rdevice-condvar
+         (and (##not (##eq? direction (macro-direction-out)))
+              (##make-rdevice-condvar raw-device)))
+        (wdevice-condvar
+         (and (##not (##eq? direction (macro-direction-in)))
+              (##make-wdevice-condvar raw-device)))
+        (uname (or uname 'raw-device)))
+
+    (define (name port)
+
+      ;; It is assumed that the thread **does not** have exclusive
+      ;; access to the port.
+
+      (##declare (not interrupts-enabled))
+      (##list uname (macro-raw-device-port-device port)))
```

Comment:
Added a type argument, and made the name return the type consed with the device for nice debugging output. 
No need to get the type from the name any more though, it's available directly from the port.